### PR TITLE
Fix TF Lite Slice kernel behaviour

### PR DESCRIPTION
**Pull request by akhtyamovpavel**
This PR fixes the incorrect behaviour of the TF Lite Slice operation.

Citing the documentation: [https://www.tensorflow.org/api_docs/python/tf/slice](https://www.tensorflow.org/api_docs/python/tf/slice)

> If `size[i]` is -1, all remaining elements in dimension `i` are included in the slice. In other words, this is equivalent to setting: `size[i] = input.dim_size(i) - begin[i]`

However, the behaviour of code was described as: `size[i] = input.dim_size(i) - 2 * begin[i]`.

This PR fixes described bug and adds tests for the following cases: `begin[i]` for `i >= 1` is 1 while the `end[i]` is -1.

### CODE REVIEW

**Comment by rthadur** [comment]
@akhtyamovpavel please sign CLA

**Comment by jdduke** [comment]
Is there a test case we can add in lite/testing/generate_examples.py?

**Comment by akhtyamovpavel** [comment]
@jdduke, I have added tests for the `generating_examples` with size -1
@rthadur, we are investigating the questions with corporate CLA. As soon as we could solve this problem (in the near future), I will give you a sign.

**Comment by akhtyamovpavel** [comment]
I signed it!

**Comment by akhtyamovpavel** [comment]
@rthadur Could you help with passing of the CLA Review? We have signed and accepted request for the Corporate CLA. Furthermore, I have changed the email of commits via `rebase` and amend commands. However, the CLA check has not been completed.

**Comment by akhtyamovpavel** [comment]
@alanchiao @jdduke We have signed the CLA. Could you review the code or notify about the announced date for the reviewing this pull request?

**Review by jdduke — DISMISSED** [review]
Thanks for the test!

**Comment by rthadur** [comment]
@akhtyamovpavel  can you please address build failures.

**Comment by akhtyamovpavel** [comment]
@rthadur I have fixed sanity build, the issue was about having too many characters in line

File: tensorflow/lite/testing/generate_examples_lib.py [hunk 1]
```diff
@@ -3758,6 +3758,25 @@ def make_slice_tests(options):
           "begin": [[0, 0], [1, 0]],
           "size": [[2, 3], [2, 2]],
       },
+      # 4-D with size -1
+      {
+          "dtype": [tf.float32, tf.int32, tf.int64],
```
Comment (jdduke) [inline tensorflow/lite/testing/generate_examples_lib.py]: For this case let's just do dtype: tffloat32, and index_type: tf.int32.

Otherwise looks great!

**Review by jdduke — CHANGES_REQUESTED** [review]

File: tensorflow/lite/testing/generate_examples_lib.py (hunk 1 above)
Comment (akhtyamovpavel) [reply tensorflow/lite/testing/generate_examples_lib.py]: @jdduke, I have changed this

**Review by jdduke — APPROVED** [review]

**Comment by akhtyamovpavel** [comment]
@jdduke, could you restart the build or say how to get logs from the Ubuntu CC checks? It seems that the previous build has been failed for the other reasons.

**Review by jdduke — APPROVED** [review]

**Comment by jdduke** [comment]
@rthadur can you help move this through the presubmit check?

**Pull request merged**